java-checks-test-sources/src/main/files/non-compiling/checks/security/CipherBlockChainingCheck.java

```diff
@@ -0,0 +1,33 @@
+package checks.security;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.security.SecureRandom;
+import javax.crypto.spec.IvParameterSpec;
+
+class CipherBlockChainingCheck {
+
+  SecureRandom random = new SecureRandom();
+  private byte[] bytes2 = "111".getBytes();
+  {
+    random.nextBytes(bytes2);
+  }
+  IvParameterSpec iv = new IvParameterSpec(bytes2); // Compliant FN, will be fixed with SE engine
+
+  static void byteBufferNotUsedAsIv(ByteBuffer ivBuffer) {
+    final byte[] biv = new byte[16];
+    final byte[] other = new byte[16];
+    ivBuffer.get(other);
+    IvParameterSpec iv = new IvParameterSpec(biv); // Noncompliant
+  }
+
+  void unknownFunctionFromRandomIv() throws UnsupportedEncodingException {
+    byte[] iv = random.unknown(16);
+    IvParameterSpec ivParameterSpec = new IvParameterSpec(iv); // Compliant, because function cannot be resolved
+  }
+
+  void unknownFunctionIv() throws UnsupportedEncodingException {
+    byte[] iv = unknown(16); // "iv" is random thanks to SecureRandom
```

Comment:
The comment doesn't make sense to me - a remainder of copy+pasting test code from elsewhere?

Reply:
Yes, you're right. That comes from a copy and past of the compiling tests.